ballerina/Module.md

```diff
@@ -0,0 +1,127 @@
+* `<password>` is the password you use to login to Snowflake account.
+
+```ballerina
+string account_identifier = "<account_identifier>";  // Eg: "z******.europe-west4.gcp" 
+string user = "<username>";
+string password = "<password>";
+snowflake:Options options = {
+    requestGeneratedKeys: snowflake:NONE  // This should be specified
+};
+
+snowflake:Client snowflakeClient = check new (account_identifier, user, password, options);
+```
+
+You can also define `<account_identifier>`, `<username>` and `<password>` as configurable strings in your Ballerina program.
+
+### Step 3: Invoke the connector operation
+1. Use the Snowflake connector to consume all of the Snowflake applications, databases, APIs, services via the Java Database Connectivity (JDBC) API using Ballerina.
+
+    Now let’s take a look at a few sample operations.
+
+    Let’s assume,
+    - `COMPANYDB` is the database name. 
+    - `PUBLIC` is the schema name. 
+    - `EMPLOYEES` is the table name
+
+    Use the `query` operation to query data. 
+
+    Following is a sample code to query data from a table.
+
+    ```ballerina
+    public function main() returns error? {
+        sql:ParameterizedQuery sqlQuery = `SELECT * FROM COMPANYDB.PUBLIC.EMPLOYEES LIMIT 10`;
+        stream<record {}, error?> resultStream = snowflakeClient->query(sqlQuery);
+
+        check from record{} result in resultStream
+            do {
+                io:println("Full details of employee: ", result);
+            };
+    }
+    ``` 
+
+    Use the `execute` operation to perform DML and DDL operations.
+
+    Following is a sample code to insert data into a table
+
+    ```ballerina
+    public function main() returns error? {
+        sql:ParameterizedQuery sqlQuery = `INSERT INTO COMPANYDB.PUBLIC.EMPLOYEES (FirstName,
+            LastName, Company) VALUES ('Shawn', 'Jerome', 'WSO2')`;
+        _ = check snowflakeClient->execute(sqlQuery);
+    }
+    ```
+
+    Use the `batchExecute` operation to perform a batch of DML and DDL operations.
+
+    Following is a sample code to insert multiple records into a table
+
+    ```ballerina
+    public function main() returns error? {
+        var insertRecords = [
+            {
+                FirstName: "Gloria",
+                LastName: "Shania",
+                Company: "ABC"
+            }, 
+            {
+                FirstName: "Shane",
+                LastName: "Warny",
+                Company: "BCA"
+            }, 
+            {
+                FirstName: "Neo",
+                LastName: "Mark",
+                Company: "CAB"
+            }
+        ];
+
+        sql:ParameterizedQuery[] insertQueries = 
+            from var data in insertRecords
+            select `INSERT INTO COMPANYDB.PUBLIC.EMPLOYEES
+                    (FirstName, LastName, Company)
+                    VALUES (${data.FirstName}, ${data.LastName}, ${data.Company})`;
+
+        _ = check snowflakeClient->batchExecute(insertQueries);
+    }
+    ```
```

Comment:
If possible try to add a simple example for executing a stored procedure too. Just a suggestion.

Reply:
Let's add